Exception with partial content (206)

I try to serve .mp4 files, so I put an nginx in front of s3proxy to set the right `content-type` (video/mp4). Once I do this I can write the following url in the browser: http://localhost/6015_2018_2035.h264.cutted.mp4

If I don't set the `content-type` the video is downloaded instead of being reproduced in the browser (chrome in my case).

The problem is that when chrome tries to reproduce the video it sends two connections. The second one responds with a 206 (partial content) and with the right header (byte ranges) but at the end I get the following exception:

```
15:26:13.819 [qtp2032647583-21 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.WriteFlusher - update WriteFlusher@18916c5b{WRITING}:IDLE-->WRITING
15:26:13.819 [qtp2032647583-21 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.ChannelEndPoint - flushed 32768 SelectChannelEndPoint@4465ed37{/127.0.0.1:48839<->9090,Open,in,out,-,W,1/30000,HttpConnection}{io=0,kio=0,kro=1}
15:26:13.819 [qtp2032647583-21 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.WriteFlusher - update WriteFlusher@18916c5b{IDLE}:WRITING-->IDLE
15:26:13.819 [qtp2032647583-21 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG o.e.jetty.server.HttpConnection - org.eclipse.jetty.server.HttpConnection$SendCallback@56cd6ef[PROCESSING][i=null,cb=Blocker@69df07da{null}] generate: DONE (null,[p=32768,l=32768,c=32768,r=0],false)@COMMITTED
15:26:13.819 [qtp2032647583-21 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG o.e.jetty.server.HttpConnection - org.eclipse.jetty.server.HttpConnection$SendCallback@56cd6ef[PROCESSING][i=null,cb=Blocker@69df07da{null}] generate: FLUSH (null,[p=0,l=32768,c=32768,r=32768],false)@COMMITTED
15:26:13.819 [qtp2032647583-21 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.WriteFlusher - write: WriteFlusher@18916c5b{IDLE} [HeapByteBuffer@3fb28ad9[p=0,l=32768,c=32768,r=32768]={<<<\x1d\x88\x11x6=\xA4\\QR\xF7Z\xD5\xFa\xB2\x90\x94...\x8b}\x1b@\xD7\x8bF\xAb\xCb\x12\xEd\x85\x06t\x0f>>>}]
15:26:13.819 [qtp2032647583-21 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.WriteFlusher - update WriteFlusher@18916c5b{WRITING}:IDLE-->WRITING
15:26:13.819 [qtp2032647583-21 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.ChannelEndPoint - flushed 32768 SelectChannelEndPoint@4465ed37{/127.0.0.1:48839<->9090,Open,in,out,-,W,0/30000,HttpConnection}{io=0,kio=0,kro=1}
15:26:13.819 [qtp2032647583-21 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.WriteFlusher - update WriteFlusher@18916c5b{IDLE}:WRITING-->IDLE
15:26:13.819 [qtp2032647583-21 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG o.e.jetty.server.HttpConnection - org.eclipse.jetty.server.HttpConnection$SendCallback@56cd6ef[PROCESSING][i=null,cb=Blocker@69df07da{null}] generate: DONE (null,[p=32768,l=32768,c=32768,r=0],false)@COMMITTED
15:26:13.819 [qtp2032647583-21 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG o.e.jetty.server.HttpConnection - org.eclipse.jetty.server.HttpConnection$SendCallback@56cd6ef[PROCESSING][i=null,cb=Blocker@69df07da{null}] generate: FLUSH (null,[p=0,l=32768,c=32768,r=32768],false)@COMMITTED
15:26:13.819 [qtp2032647583-21 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.WriteFlusher - write: WriteFlusher@18916c5b{IDLE} [HeapByteBuffer@3fb28ad9[p=0,l=32768,c=32768,r=32768]={<<<\x17\x88\x8eC\xA4C\xC5\x85\xB3\x88\x1a\xA5\xB4\xDft0\xAb...\xDd\xA5\xFd\xC1\xC9\xEb\x0b\xD7\xB9\x0f\xF3\xD329I>>>}]
15:26:13.819 [qtp2032647583-21 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.WriteFlusher - update WriteFlusher@18916c5b{WRITING}:IDLE-->WRITING
15:26:13.819 [qtp2032647583-21 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.ChannelEndPoint - flushed 32768 SelectChannelEndPoint@4465ed37{/127.0.0.1:48839<->9090,Open,in,out,-,W,0/30000,HttpConnection}{io=0,kio=0,kro=1}
15:26:13.820 [qtp2032647583-21 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.WriteFlusher - update WriteFlusher@18916c5b{IDLE}:WRITING-->IDLE

...
... more than 4000 lines with the same trace
...

15:26:14.851 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.WriteFlusher - write: WriteFlusher@1703ec5e{IDLE} [HeapByteBuffer@3fb28ad9[p=0,l=32768,c=32768,r=32768]={<<<T\xEe\x07T\xE9e\x8e\xBce\x0f\x18Oa~\xBa}\xB1...\xEa\xFb\x08\xF3e\xE1ydDe?\xC9\x04\xF4\xD6>>>}]
15:26:14.851 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.WriteFlusher - update WriteFlusher@1703ec5e{WRITING}:IDLE-->WRITING
15:26:14.851 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.ChannelEndPoint - flushed 32768 SelectChannelEndPoint@4d0efa12{/127.0.0.1:48843<->9090,Open,in,out,-,W,0/30000,HttpConnection}{io=0,kio=0,kro=1}
15:26:14.851 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.WriteFlusher - update WriteFlusher@1703ec5e{IDLE}:WRITING-->IDLE
15:26:14.851 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG o.e.jetty.server.HttpConnection - org.eclipse.jetty.server.HttpConnection$SendCallback@60b9d7b1[PROCESSING][i=null,cb=Blocker@35eea7f5{null}] generate: DONE (null,[p=32768,l=32768,c=32768,r=0],false)@COMMITTED
15:26:14.851 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG o.e.jetty.server.HttpConnection - org.eclipse.jetty.server.HttpConnection$SendCallback@60b9d7b1[PROCESSING][i=null,cb=Blocker@35eea7f5{null}] generate: FLUSH (null,[p=0,l=32768,c=32768,r=32768],false)@COMMITTED
15:26:14.851 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.WriteFlusher - write: WriteFlusher@1703ec5e{IDLE} [HeapByteBuffer@3fb28ad9[p=0,l=32768,c=32768,r=32768]={<<<JI\x87\x1a\xF5\xDa8\xD5e\xF0\xEem\xFd\xC12\xEc...\x9b\xEf\xA9H\xF6\xD6*\xB0~\xD7R]1-\x1b>>>}]
15:26:14.851 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.WriteFlusher - update WriteFlusher@1703ec5e{WRITING}:IDLE-->WRITING
15:26:14.853 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.WriteFlusher - write exception
org.eclipse.jetty.io.EofException: null
    at org.eclipse.jetty.io.ChannelEndPoint.flush(ChannelEndPoint.java:192) ~[jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.io.WriteFlusher.flush(WriteFlusher.java:408) ~[jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.io.WriteFlusher.write(WriteFlusher.java:302) ~[jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.io.AbstractEndPoint.write(AbstractEndPoint.java:129) [jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpConnection$SendCallback.process(HttpConnection.java:690) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.util.IteratingCallback.processing(IteratingCallback.java:246) [jetty-util-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.util.IteratingCallback.iterate(IteratingCallback.java:208) [jetty-util-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpConnection.send(HttpConnection.java:480) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpChannel.sendResponse(HttpChannel.java:768) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpChannel.write(HttpChannel.java:801) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpOutput.write(HttpOutput.java:147) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpOutput.write(HttpOutput.java:140) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpOutput.write(HttpOutput.java:355) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at com.google.common.io.ByteStreams.copy(ByteStreams.java:179) [guava-16.0.1.jar:na]
    at org.gaul.s3proxy.S3ProxyHandler.handleGetBlob(S3ProxyHandler.java:979) [classes/:na]
    at org.gaul.s3proxy.S3ProxyHandler.doHandle(S3ProxyHandler.java:392) [classes/:na]
    at org.gaul.s3proxy.S3ProxyHandler.handle(S3ProxyHandler.java:201) [classes/:na]
    at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.Server.handle(Server.java:499) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:310) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540) [jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635) [jetty-util-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555) [jetty-util-9.2.11.v20150529.jar:9.2.11.v20150529]
    at java.lang.Thread.run(Thread.java:745) [na:1.8.0_45-internal]
Caused by: java.io.IOException: Broken pipe
    at sun.nio.ch.FileDispatcherImpl.write0(Native Method) ~[na:1.8.0_45-internal]
    at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:47) ~[na:1.8.0_45-internal]
    at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93) ~[na:1.8.0_45-internal]
    at sun.nio.ch.IOUtil.write(IOUtil.java:65) ~[na:1.8.0_45-internal]
    at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471) ~[na:1.8.0_45-internal]
    at org.eclipse.jetty.io.ChannelEndPoint.flush(ChannelEndPoint.java:170) ~[jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    ... 24 common frames omitted
15:26:14.853 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.WriteFlusher - update WriteFlusher@1703ec5e{IDLE}:WRITING-->IDLE
15:26:14.853 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG o.e.jetty.server.HttpConnection - org.eclipse.jetty.server.HttpConnection$SendCallback@60b9d7b1[PROCESSING][i=null,cb=Blocker@35eea7f5{null}] generate: FLUSH (null,[p=0,l=32768,c=32768,r=32768],true)@COMPLETING
15:26:14.853 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.WriteFlusher - write: WriteFlusher@1703ec5e{IDLE} [HeapByteBuffer@3fb28ad9[p=0,l=32768,c=32768,r=32768]={<<<JI\x87\x1a\xF5\xDa8\xD5e\xF0\xEem\xFd\xC12\xEc...\x9b\xEf\xA9H\xF6\xD6*\xB0~\xD7R]1-\x1b>>>}]
15:26:14.853 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.WriteFlusher - update WriteFlusher@1703ec5e{WRITING}:IDLE-->WRITING
15:26:14.853 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.WriteFlusher - write exception
org.eclipse.jetty.io.EofException: null
    at org.eclipse.jetty.io.ChannelEndPoint.flush(ChannelEndPoint.java:192) ~[jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.io.WriteFlusher.flush(WriteFlusher.java:408) ~[jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.io.WriteFlusher.write(WriteFlusher.java:302) ~[jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.io.AbstractEndPoint.write(AbstractEndPoint.java:129) [jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpConnection$SendCallback.process(HttpConnection.java:690) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.util.IteratingCallback.processing(IteratingCallback.java:246) [jetty-util-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.util.IteratingCallback.iterate(IteratingCallback.java:208) [jetty-util-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpConnection.send(HttpConnection.java:480) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpChannel.sendResponse(HttpChannel.java:768) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpChannel.write(HttpChannel.java:801) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpOutput.write(HttpOutput.java:147) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpOutput.write(HttpOutput.java:140) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpOutput.close(HttpOutput.java:171) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.gaul.s3proxy.S3ProxyHandler.handleGetBlob(S3ProxyHandler.java:981) [classes/:na]
    at org.gaul.s3proxy.S3ProxyHandler.doHandle(S3ProxyHandler.java:392) [classes/:na]
    at org.gaul.s3proxy.S3ProxyHandler.handle(S3ProxyHandler.java:201) [classes/:na]
    at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.Server.handle(Server.java:499) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:310) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540) [jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635) [jetty-util-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555) [jetty-util-9.2.11.v20150529.jar:9.2.11.v20150529]
    at java.lang.Thread.run(Thread.java:745) [na:1.8.0_45-internal]
Caused by: java.io.IOException: Broken pipe
    at sun.nio.ch.FileDispatcherImpl.write0(Native Method) ~[na:1.8.0_45-internal]
    at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:47) ~[na:1.8.0_45-internal]
    at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93) ~[na:1.8.0_45-internal]
    at sun.nio.ch.IOUtil.write(IOUtil.java:65) ~[na:1.8.0_45-internal]
    at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471) ~[na:1.8.0_45-internal]
    at org.eclipse.jetty.io.ChannelEndPoint.flush(ChannelEndPoint.java:170) ~[jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    ... 23 common frames omitted
15:26:14.853 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.WriteFlusher - update WriteFlusher@1703ec5e{IDLE}:WRITING-->IDLE
15:26:14.854 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.server.HttpOutput - 
org.eclipse.jetty.io.EofException: null
    at org.eclipse.jetty.io.ChannelEndPoint.flush(ChannelEndPoint.java:192) ~[jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.io.WriteFlusher.flush(WriteFlusher.java:408) ~[jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.io.WriteFlusher.write(WriteFlusher.java:302) ~[jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.io.AbstractEndPoint.write(AbstractEndPoint.java:129) ~[jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpConnection$SendCallback.process(HttpConnection.java:690) ~[jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.util.IteratingCallback.processing(IteratingCallback.java:246) ~[jetty-util-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.util.IteratingCallback.iterate(IteratingCallback.java:208) ~[jetty-util-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpConnection.send(HttpConnection.java:480) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpChannel.sendResponse(HttpChannel.java:768) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpChannel.write(HttpChannel.java:801) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpOutput.write(HttpOutput.java:147) ~[jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpOutput.write(HttpOutput.java:140) ~[jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpOutput.close(HttpOutput.java:171) ~[jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.gaul.s3proxy.S3ProxyHandler.handleGetBlob(S3ProxyHandler.java:981) [classes/:na]
    at org.gaul.s3proxy.S3ProxyHandler.doHandle(S3ProxyHandler.java:392) [classes/:na]
    at org.gaul.s3proxy.S3ProxyHandler.handle(S3ProxyHandler.java:201) [classes/:na]
    at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.Server.handle(Server.java:499) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:310) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540) [jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635) [jetty-util-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555) [jetty-util-9.2.11.v20150529.jar:9.2.11.v20150529]
    at java.lang.Thread.run(Thread.java:745) [na:1.8.0_45-internal]
Caused by: java.io.IOException: Broken pipe
    at sun.nio.ch.FileDispatcherImpl.write0(Native Method) ~[na:1.8.0_45-internal]
    at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:47) ~[na:1.8.0_45-internal]
    at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93) ~[na:1.8.0_45-internal]
    at sun.nio.ch.IOUtil.write(IOUtil.java:65) ~[na:1.8.0_45-internal]
    at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471) ~[na:1.8.0_45-internal]
    at org.eclipse.jetty.io.ChannelEndPoint.flush(ChannelEndPoint.java:170) ~[jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    ... 23 common frames omitted
15:26:14.854 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG o.eclipse.jetty.io.AbstractEndPoint - onClose SelectChannelEndPoint@4d0efa12{/127.0.0.1:48843<->9090,CLOSED,in,out,-,-,3/30000,HttpConnection}{io=0,kio=0,kro=1}
15:26:14.854 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.ChannelEndPoint - close SelectChannelEndPoint@4d0efa12{/127.0.0.1:48843<->9090,CLOSED,in,out,-,-,3/30000,HttpConnection}{io=0,kio=0,kro=1}
15:26:14.854 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.io.SelectorManager - Destroyed SelectChannelEndPoint@4d0efa12{/127.0.0.1:48843<->9090,CLOSED,ISHUT,OSHUT,-,-,3/30000,HttpConnection}{io=0,kio=-1,kro=-1}
15:26:14.854 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG o.e.jetty.io.AbstractConnection - onClose HttpConnection@711dd5b5{FILLING}
15:26:14.854 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG o.eclipse.jetty.io.AbstractEndPoint - onClose SelectChannelEndPoint@4d0efa12{/127.0.0.1:48843<->9090,CLOSED,ISHUT,OSHUT,-,-,3/30000,HttpConnection}{io=0,kio=-1,kro=-1}
15:26:14.854 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.server.HttpChannel - 
org.eclipse.jetty.io.EofException: null
    at org.eclipse.jetty.io.ChannelEndPoint.flush(ChannelEndPoint.java:192) ~[jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.io.WriteFlusher.flush(WriteFlusher.java:408) ~[jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.io.WriteFlusher.write(WriteFlusher.java:302) ~[jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.io.AbstractEndPoint.write(AbstractEndPoint.java:129) ~[jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpConnection$SendCallback.process(HttpConnection.java:690) ~[jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.util.IteratingCallback.processing(IteratingCallback.java:246) ~[jetty-util-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.util.IteratingCallback.iterate(IteratingCallback.java:208) ~[jetty-util-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpConnection.send(HttpConnection.java:480) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpChannel.sendResponse(HttpChannel.java:768) ~[jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpChannel.write(HttpChannel.java:801) ~[jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpOutput.write(HttpOutput.java:147) ~[jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpOutput.write(HttpOutput.java:140) ~[jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpOutput.write(HttpOutput.java:355) ~[jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at com.google.common.io.ByteStreams.copy(ByteStreams.java:179) ~[guava-16.0.1.jar:na]
    at org.gaul.s3proxy.S3ProxyHandler.handleGetBlob(S3ProxyHandler.java:979) ~[classes/:na]
    at org.gaul.s3proxy.S3ProxyHandler.doHandle(S3ProxyHandler.java:392) ~[classes/:na]
    at org.gaul.s3proxy.S3ProxyHandler.handle(S3ProxyHandler.java:201) ~[classes/:na]
    at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97) ~[jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.Server.handle(Server.java:499) ~[jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:310) ~[jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257) [jetty-server-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540) [jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635) [jetty-util-9.2.11.v20150529.jar:9.2.11.v20150529]
    at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555) [jetty-util-9.2.11.v20150529.jar:9.2.11.v20150529]
    at java.lang.Thread.run(Thread.java:745) [na:1.8.0_45-internal]
Caused by: java.io.IOException: Broken pipe
    at sun.nio.ch.FileDispatcherImpl.write0(Native Method) ~[na:1.8.0_45-internal]
    at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:47) ~[na:1.8.0_45-internal]
    at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93) ~[na:1.8.0_45-internal]
    at sun.nio.ch.IOUtil.write(IOUtil.java:65) ~[na:1.8.0_45-internal]
    at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471) ~[na:1.8.0_45-internal]
    at org.eclipse.jetty.io.ChannelEndPoint.flush(ChannelEndPoint.java:170) ~[jetty-io-9.2.11.v20150529.jar:9.2.11.v20150529]
    ... 24 common frames omitted
15:26:14.854 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG o.e.jetty.server.HttpChannelState - HttpChannelState@50db53b7{s=DISPATCHED i=true a=null} unhandle DISPATCHED
15:26:14.854 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.http.HttpParser - close HttpParser{s=END,0 of 0}
15:26:14.854 [qtp2032647583-18 - /videos/6015_2018_2035.h264.cutted.mp4] DEBUG org.eclipse.jetty.http.HttpParser - END --> CLOSED
15:26:14.854 [qtp2032647583-18] DEBUG org.eclipse.jetty.server.HttpChannel - HttpChannelOverHttp@3f99f0a6{r=1,c=false,a=IDLE,uri=-} handle exit, result COMPLETE
15:26:14.854 [qtp2032647583-18] DEBUG org.eclipse.jetty.http.HttpParser - atEOF HttpParser{s=CLOSED,0 of 0}
15:26:14.854 [qtp2032647583-18] DEBUG org.eclipse.jetty.http.HttpParser - parseNext s=CLOSED HeapByteBuffer@40635532[p=0,l=0,c=0,r=0]={<<<>>>}
15:26:14.854 [qtp2032647583-18] DEBUG o.e.jetty.io.AbstractConnection - FILLING-->IDLE HttpConnection@711dd5b5{IDLE}
```

--- a/src/main/java/org/gaul/s3proxy/S3ProxyHandler.java
+++ b/src/main/java/org/gaul/s3proxy/S3ProxyHandler.java
@@ -978,7 +978,15 @@ final class S3ProxyHandler extends AbstractHandler {
         }
 
         response.setStatus(status);
+
         addMetadataToResponse(response, blob.getMetadata());
+        Collection<String> contentRanges =
+                blob.getAllHeaders().get(HttpHeaders.CONTENT_RANGE);
+        if (!contentRanges.isEmpty()) {
+            response.addHeader(HttpHeaders.CONTENT_RANGE,
+                    contentRanges.iterator().next());
+        }
+
         try (InputStream is = blob.getPayload().openStream();
              OutputStream os = response.getOutputStream()) {
             ByteStreams.copy(is, os);

